webcam.go

```diff
@@ -106,6 +106,26 @@ func (w *Webcam) GetSupportedFrameSizes(f PixelFormat) []FrameSize {
 	return result
 }
 
+// GetSupportedFramerates returns supported frame rates for a given image format and frame size.
+func (w *Webcam) GetSupportedFramerates(fp PixelFormat, width uint32, height uint32) []FrameRate {
+	var result []FrameRate
+	var index uint32
+	var err error
+
+	// keep incrementing the index value until we get an EINVAL error
+	index = 0
+	for err == nil {
```

Comment:
Ah, I see what you mean about the for loop. Yeah it seems cleaner to do this all on one row like elsewhere in the code.

Reply:
So when I read code like this, I think usage of index in a for loop is for iterating a known, discrete range. I like a while loop better since to me it reads, "call getFrameInterval while no error occurs" as opposed to "for each index, starting at 0 and while no errors occurs, call getFrameInterval, then check error again, and if okay, increment the index". I think lexically it's easier to see the conditions but this is fairly subjective on seeing some more complex versions that are harder to read.

Reply:
Yeah I can definitely see that being more clear.

Reply:
Ok keeping `err == nil` loop.